### PR TITLE
Add new rule "require-valid-svg"

### PR DESCRIPTION
**Pull request by KamiKillertO**
This is an integration for the rule described in #1507

### CODE REVIEW

File: test/unit/rules/require-valid-svg-test.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    '<button aria-label="sort ascending to descending"><svg aria-hidden="true"></svg></button>',
+    '<button aria-label="sort ascending to descending"><svg ...attributes></svg></button>',
+  ],
+
+  bad: [
+    {
+      template: '<svg role="image"></svg>',
+
+      result: {
+        message: 'An `<svg>` element with role `image` should have a none empty alternative text',
```
Comment (MelSumner) [inline test/unit/rules/require-valid-svg-test.js:25]:
```suggestion
        message: 'An `<svg>` element with role `image` should have valid alternative text',
```

File: test/unit/rules/require-valid-svg-test.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        source: '<svg role="image"></svg>',
+        line: 1,
+        column: 0,
+      },
+    },
+    {
+      template: '<svg role="image" alt=""></svg>',
+
+      result: {
+        message: 'An `<svg>` element with role `image` should have a none empty alternative text',
```
Comment (MelSumner) [inline test/unit/rules/require-valid-svg-test.js:35]:
```suggestion
        message: 'An `<svg>` element with role `image` should have valid alternative text',
```

File: test/unit/rules/require-valid-svg-test.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        line: 1,
+        column: 0,
+      },
+    },
+    {
+      template: '<svg role="image" alt="valid alternative text"><g></g></svg>',
+
+      result: {
+        message:
+          'An `<svg>` element with role `image` should contains contain a none empty `<title>` element',
```
Comment (MelSumner) [inline test/unit/rules/require-valid-svg-test.js:46]:
```suggestion
          'An `<svg>` element with role `image` should contains contain a valid `<title>` element',
```

File: test/unit/rules/require-valid-svg-test.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        line: 1,
+        column: 0,
+      },
+    },
+    {
+      template: '<svg role="image" alt="valid alternative text"><title>  </title></svg>',
+
+      result: {
+        message:
+          'An `<svg>` element with role `image` should contains contain a none empty `<title>` element',
```
Comment (MelSumner) [inline test/unit/rules/require-valid-svg-test.js:57]:
```suggestion
          'An `<svg>` element with role `image` should contains contain a valid `<title>` element',
```

File: test/unit/rules/require-valid-svg-test.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        source: '<svg role="image" alt="valid alternative text"><title>  </title></svg>',
+        line: 1,
+        column: 0,
+      },
+    },
+    {
+      template: '<svg role="image"><title>Image description</title><g></g></svg>',
+
+      result: {
+        message: 'An `<svg>` element with role `image` should have a none empty alternative text',
```
Comment (MelSumner) [inline test/unit/rules/require-valid-svg-test.js:67]:
```suggestion
        message: 'An `<svg>` element with role `image` should have valid alternative text',
```

File: test/unit/rules/require-valid-svg-test.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        line: 1,
+        column: 0,
+      },
+    },
+    {
+      template: '<svg alt></svg>',
+
+      result: {
+        message:
+          'An `<svg>` element without role `image` should not have the `alt` attribute defined',
```
Comment (MelSumner) [inline test/unit/rules/require-valid-svg-test.js:78]:
```suggestion
          'An `<svg>` element without `role="image"` should not have the `alt` attribute defined',
```

File: test/unit/rules/require-valid-svg-test.js [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        line: 1,
+        column: 8,
+      },
+    },
+    {
+      template: '<button><svg aria-hidden="true"></svg></button>',
+
+      result: {
+        message:
+          'Parent tag of an `<svg>` element should have an none empty `aria-label` attribute',
```
Comment (MelSumner) [inline test/unit/rules/require-valid-svg-test.js:100]:
```suggestion
          'Parent tag of an `<svg>` element should have a valid `aria-label` attribute',
```

File: docs/rule/require-valid-svg.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+```hbs
+<button aria-label="sort ascending to descending">
+  <svg aria-hidden="true">...</svg>
+</button>
+```
+
+## References
+
+> All non-text content that is presented to the user has a text alternative that serves the equivalent purpose, with few exceptions. See: [WCAG 1.1.1 - Non-text Content](https://www.w3.org/TR/WCAG21/#non-text-content)
```
Comment (MelSumner) [inline docs/rule/require-valid-svg.md:53]:
```suggestion
- All non-text content that is presented to the user has a text alternative that serves the equivalent purpose, with few exceptions. See: [WCAG 1.1.1 - Non-text Content](https://www.w3.org/TR/WCAG21/#non-text-content)
```

**Review by MelSumner — CHANGES_REQUESTED** [review]
Please also add this new rule to the `a11y.js` config file. Thank you!

**Comment by KamiKillertO** [comment]
Thanks, @MelSumner for the review, but do you think we should keep this rule or as mentioned in this [comment](https://github.com/ember-template-lint/ember-template-lint/issues/1507#issuecomment-702603780) split the rule into multiples rules.

File: lib/rules/require-valid-svg.js [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+        let parent = getParentLinkOrButton(path);
+        if (parent) {
+          // Check '...attributes' ?
+          if (
+            AstNodeInfo.hasAttributeValue(node, 'aria-hidden', 'true') === false &&
+            AstNodeInfo.hasAttribute(node, '...attributes') === false
+          ) {
+            return this.logNode({
+              message:
+                'An `<svg>` element inside a `<a>` or `<button>` element should have the `aria-hidden` attribute set to true',
```
Comment (lifeart) [inline lib/rules/require-valid-svg.js:59]: wondering, could we add fixer for it?

**Pull request closed without merging**